exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentContext.java

```diff
@@ -17,483 +17,216 @@
-    throws ExecutionSetupException {
-    super(funcRegistry);
-    this.context = dbContext;
-    this.queryContext = queryContext;
-    this.connection = connection;
-    this.accountingUserConnection = new AccountingUserConnection(connection, sendingAccountor, statusHandler);
-    this.fragment = fragment;
-    contextInformation = new ContextInformation(fragment.getCredentials(), fragment.getContext());
-
-    logger.debug("Getting initial memory allocation of {}", fragment.getMemInitial());
-    logger.debug("Fragment max allocation: {}", fragment.getMemMax());
-
-    final OptionList list;
-    if (!fragment.hasOptionsJson() || fragment.getOptionsJson().isEmpty()) {
-      list = new OptionList();
-    } else {
-      try {
-        list = dbContext.getLpPersistence().getMapper().readValue(fragment.getOptionsJson(), OptionList.class);
-      } catch (final Exception e) {
-        throw new ExecutionSetupException("Failure while reading plan options.", e);
-      }
-    }
-    fragmentOptions = new FragmentOptionManager(context.getOptionManager(), list);
-
-    executionControls = new ExecutionControls(fragmentOptions, dbContext.getEndpoint());
-
-    // Add the fragment context to the root allocator.
-    // The QueryManager will call the root allocator to recalculate all the memory limits for all the fragments
-    try {
-      allocator = context.getAllocator().newChildAllocator(
-          "frag:" + QueryIdHelper.getFragmentId(fragment.getHandle()),
-          fragment.getMemInitial(),
-          fragment.getMemMax());
-      Preconditions.checkNotNull(allocator, "Unable to acuqire allocator");
-    } catch (final OutOfMemoryException e) {
-      throw UserException.memoryError(e)
-        .addContext("Fragment", getHandle().getMajorFragmentId() + ":" + getHandle().getMinorFragmentId())
-        .build(logger);
-    } catch(final Throwable e) {
-      throw new ExecutionSetupException("Failure while getting memory allocator for fragment.", e);
-    }
-
-    stats = new FragmentStats(allocator, fragment.getAssignment());
-    bufferManager = new BufferManagerImpl(this.allocator);
-    constantValueHolderCache = Maps.newHashMap();
-  }
+  FunctionImplementationRegistry getFunctionRegistry();
 
   /**
-   * TODO: Remove this constructor when removing the SimpleRootExec (DRILL-2097). This is kept only to avoid modifying
-   * the long list of test files.
+   * Returns a read-only version of the session options.
+   * @return the session options
    */
-  public FragmentContext(DrillbitContext dbContext, PlanFragment fragment, UserClientConnection connection,
-      FunctionImplementationRegistry funcRegistry) throws ExecutionSetupException {
-    this(dbContext, fragment, null, connection, funcRegistry);
-  }
+  OptionSet getOptionSet();
 
-  public OptionManager getOptions() {
-    return fragmentOptions;
-  }
+  PhysicalPlanReader getPlanReader();
```

Comment:
Only in network version. Actually, probably should not be part of the fragment context itself, but rather part of the mechanism that boots up a fragment. (Once the fragment is running, we won't read the plan again.)

Reply:
Yeah your right, it's only used by the FragmentExecutor. Removed it from the interface.

Reply:
Changed again. Moved this into ExecutorFragmentContext.